spec/core_ext/class/hierarchy_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'socket' # for a more interesting hierarchy for specs
+
+describe Class do
+  it "#hierarchy" do
+    expect(IO.hierarchy).to eq(
+      BasicSocket => {
+        Socket     => {},
+        IPSocket   => {TCPSocket => {TCPServer => {}}, UDPSocket => {}},
+        UNIXSocket => {UNIXServer => {}}
+      },
+      File => {}
+    )
+  end
+
+  it "#lineage" do
+    expect(TCPServer.lineage).to eq [TCPServer, TCPSocket, IPSocket, BasicSocket, IO, Object, BasicObject]
```

Comment:
I think self should be removed from `lineage`, because it doesn't exist in `hierarchy`, but thoughts?